File: library/src/main/java/com/nasahapps/iaphelper/SkuItem.java
package com.nasahapps.iaphelper;

import android.support.annotation.NonNull;

import com.android.billingclient.api.SkuDetails;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Created by hhasan on 8/22/17.
 */

class SkuItem implements Comparable<SkuItem> {

    SkuDetails skuDetails;

    public SkuItem(SkuDetails details) {
        this.skuDetails = details;
    }

    public static List<SkuItem> listFromSkuDetailsList(List<SkuDetails> detailsList) {
        List<SkuItem> list = new ArrayList<>();
        for (SkuDetails details : detailsList) {
            list.add(new SkuItem(details));
        }
        Collections.sort(list);
        return list;
    }

    public static String[] getArrayOfTitles(List<SkuItem> items) {
        String[] array = new String[items.size()];
        for (int i = 0; i < items.size(); i++) {
            array[i] = items.get(i).getTitle();
        }
        return array;
    }

    public String getTitle() {
        if (skuDetails.getTitle().contains("$1 USD")) {
            return "$1 USD";
        } else if (skuDetails.getTitle().contains("$2 USD")) {
            return "$2 USD";
        } else if (skuDetails.getTitle().contains("$5 USD")) {
            return "$5 USD";
        } else if (skuDetails.getTitle().contains("$10 USD")) {
            return "$10 USD";
        } else if (skuDetails.getTitle().contains("$20 USD")) {
            return "$20 USD";
        } else if (skuDetails.getTitle().contains("$50 USD")) {
            return "$50 USD";
        } else if (skuDetails.getTitle().contains("$100 USD")) {
            return "$100 USD";
        } else {
            return skuDetails.getTitle();
        }
    }

    @Override
    public int compareTo(@NonNull SkuItem skuItem) {
        if (this.skuDetails.getPriceAmountMicros() < skuItem.skuDetails.getPriceAmountMicros()) {
            return -1;
        } else if (this.skuDetails.getPriceAmountMicros() > skuItem.skuDetails.getPriceAmountMicros()) {
            return 1;
        } else {
            return 0;
        }
    }
}

File: library/src/main/java/com/nasahapps/iaphelper/SkuItem.kt
package com.nasahapps.iaphelper

import com.android.billingclient.api.SkuDetails

/**
 * Created by hhasan on 8/22/17.
 */

internal data class SkuItem(var skuDetails: SkuDetails) : Comparable<SkuItem> {

    val title: String
        get() = when {
            skuDetails.title.contains("$1 USD") -> "$1 USD"
            skuDetails.title.contains("$2 USD") -> "$2 USD"
            skuDetails.title.contains("$5 USD") -> "$5 USD"
            skuDetails.title.contains("$10 USD") -> "$10 USD"
            skuDetails.title.contains("$20 USD") -> "$20 USD"
            skuDetails.title.contains("$50 USD") -> "$50 USD"
            skuDetails.title.contains("$100 USD") -> "$100 USD"
            else -> skuDetails.title
        }

    override fun compareTo(other: SkuItem): Int {
        return when {
            this.skuDetails.priceAmountMicros < other.skuDetails.priceAmountMicros -> -1
            this.skuDetails.priceAmountMicros > other.skuDetails.priceAmountMicros -> 1
            else -> 0
        }
    }

    companion object {

        fun listFromSkuDetailsList(detailsList: List<SkuDetails>): List<SkuItem> {
            return detailsList.map { SkuItem(it) }.sorted()
        }

        fun getArrayOfTitles(items: List<SkuItem>): Array<String> {
            return items.map { it.title }.toTypedArray()
        }

    }
}
